scripts/publish.ps1

```diff
@@ -0,0 +1,75 @@
+$ErrorActionPreference = "Stop"
+$InformationPreference = "Continue"
+$WarningPreference = "SilentlyContinue"
+
+Trap {
+    Write-Error $_.InvocationInfo.ScriptName -ErrorAction Continue
+    $line = "$($_.InvocationInfo.ScriptLineNumber): $($_.InvocationInfo.Line)"
+    Write-Error $line -ErrorAction Continue
+    Write-Error $_
+}
+
+function CommandAliasFunction {
+    Write-Information ""
+    Write-Information "$args"
+    $cmd, $args = $args
+    & "$cmd" $args
+    if ($LASTEXITCODE) {
+        throw "Exception Occured"
+    }
+    Write-Information ""
+}
+
+Set-Alias -Name ce -Value CommandAliasFunction -Scope script
+
+$AwsAccountId = $(aws sts get-caller-identity --output text --query Account)
+$EcrRegistry = "$AwsAccountId.dkr.ecr.$env:AWS_DEFAULT_REGION.amazonaws.com"
+$Image = "$EcrRegistry/$env:INPUT_ECR_REPOSITORY" + ":" + "$env:IMAGE_VERSION"
+$DockerfilePath = "$env:INPUT_DOCKERFILE_DIR_PATH"
+
+try {
+    $ImageCount = aws ecr list-images --repository-name $env:INPUT_ECR_REPOSITORY | jq '.imageIds | unique_by(.imageDigest) | length'
+    $ImageTags = aws ecr describe-images --repository-name $env:INPUT_ECR_REPOSITORY --max-items $ImageCount --query 'imageDetails[*].imageTags[0]'
+    $removeImage = $true
+    if ($ImageTags -like "*$env:IMAGE_VERSION*") {
+        Write-Output "Image already pushed. Skipping docker push"
+        $removeImage = $false
```

Comment:
Naveen, because we exit immediately out of this block, `$removeImage` will never be set to false. This is OK but above, please initialize it to `$true.

Reply:
done